tests/unit/Makefile

```diff
@@ -1,68 +1,69 @@
+GTEST_HEADERS = $(GTEST_DIR)/include/gtest/*.h \
+                $(GTEST_DIR)/include/gtest/internal/*.h
 
-algebraic_expression: test_algebraic_expression.o
-	$(CC) $(CFLAGS) -o test_algebraic_expression test_algebraic_expression.o  $(DEPS) $(LDFLAGS)
+# RedisGraph flags and libraries
 
-tuples_iter: test_tuples_iter.o
-	$(CC) $(CFLAGS) -o test_tuples_iter test_tuples_iter.o  $(DEPS) $(LDFLAGS)
+# TODO - unused right now (perhaps always?)
+# CFLAGS:=$(CFLAGS)
 
-filter_tree: test_filter_tree.o
-	$(CC) $(CFLAGS) -o test_filter_tree test_filter_tree.o  $(DEPS) $(LDFLAGS)
+LDFLAGS :=  -lc -lm -ldl -lpthread
+CC_OBJECTS:=$(CC_OBJECTS)
+LIBRMUTIL=../../src/rmutil/librmutil.a
+LIBTRIEMAP=../../src/util/triemap/libtriemap.a
+LIBGRAPHBLAS=../../deps/GraphBLAS/build/libgraphblas.a
 
-value: test_value.o
-	$(CC) $(CFLAGS)  -o test_value test_value.o  $(DEPS) $(LDFLAGS)
+LIBS=$(LIBTRIEMAP) $(LIBRMUTIL) $(LIBGRAPHBLAS)
+DEPS=$(CC_OBJECTS) $(LIBS)
 
-arithmetic_expression: test_arithmetic_expression.o
-	$(CC) $(CFLAGS)  -o test_arithmetic_expression test_arithmetic_expression.o  $(DEPS) $(LDFLAGS)
+# Test compilation targets
+TESTS = test_value test_graph
 
-resultset_record: test_resultset_record.o
-	$(CC) $(CFLAGS)  -o test_resultset_record test_resultset_record.o  $(DEPS) $(LDFLAGS)
+run_tests: $(TESTS)
+	@(sh -c ./test_value)
+	@(sh -c ./test_graph)
 
-build: graph algebraic_expression tuples_iter filter_tree value arithmetic_expression resultset_record
+all : $(TESTS) run_tests
 
-test_graph:
-	@(sh -c ./test_graph)
-.PHONY: test_graph
+clean :
+	rm -f $(TESTS) gtest.a gtest_main.a *.o
 
-test_algebraic_expression:
-	@(sh -c ./test_algebraic_expression)
-.PHONY: test_algebraic_expression
+# Google Test compiler targets
+GTEST_SRCS_ = $(GTEST_DIR)/src/*.cc $(GTEST_DIR)/src/*.h $(GTEST_HEADERS)
 
-test_tuples_iter:
-	@(sh -c ./test_tuples_iter)
-.PHONY: test_tuples_iter
+gtest-all.o : $(GTEST_SRCS_)
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c \
+            $(GTEST_DIR)/src/gtest-all.cc
 
-test_filter_tree:
-	@(sh -c ./test_filter_tree)
-.PHONY: test_filter_tree
+gtest_main.o : $(GTEST_SRCS_)
+	$(CXX) $(CPPFLAGS) -I$(GTEST_DIR) $(CXXFLAGS) -c \
+            $(GTEST_DIR)/src/gtest_main.cc
 
-test_value:
-	@(sh -c ./test_value)
-.PHONY: test_value
+gtest.a : gtest-all.o
+	$(AR) $(ARFLAGS) $@ $^
+
+gtest_main.a : gtest-all.o gtest_main.o
+	$(AR) $(ARFLAGS) $@ $^
 
-test_arithmetic_expression:
-	@(sh -c ./test_arithmetic_expression)
-.PHONY: test_arithmetic_expression
 
-test_resultset_record:
-	@(sh -c ./test_resultset_record)
-.PHONY: test_resultset_record
+# RedisGraph compiler targets
+test_value: test_value.o gtest_main.a $(DEPS)
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) $^ -o $@
 
-test: test_graph test_algebraic_expression test_tuples_iter test_filter_tree test_value test_arithmetic_expression test_resultset_record
+test_graph: test_graph.o gtest_main.a $(DEPS)
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) $^ -o $@
```

Comment:
While I'm using explicit rules right now, I think it would be easy to switch to a pattern rule to compile and run all test cases.

Reply:
That would be much nicer.